Sources/Nimble/Utils/Difference.swift

```diff
@@ -49,12 +49,29 @@ private struct Differ {
             if let expectedDict = expected as? Dictionary<AnyHashable, Any>,
                 let receivedDict = received as? Dictionary<AnyHashable, Any> {
                 var resultLines: [Line] = []
-                expectedDict.keys.forEach { key in
+                let missingKeys = Set(expectedDict.keys).subtracting(receivedDict.keys)
+                let extraKeys = Set(receivedDict.keys).subtracting(expectedDict.keys)
+                let commonKeys = Set(receivedDict.keys).intersection(expectedDict.keys)
+                commonKeys.forEach { key in
                     let results = diffLines(expectedDict[key], receivedDict[key], level: level + 1)
                     if !results.isEmpty {
                         resultLines.append(Line(contents: "Key \(key.description):", indentationLevel: level, canBeOrdered: true, children: results))
                     }
                 }
+                if (!missingKeys.isEmpty) {
+                    var missingKeyPairs: [Line] = []
+                    missingKeys.forEach { key in
+                        missingKeyPairs.append(Line(contents: "\(key.description): \(String(describing: expectedDict[key]))", indentationLevel: level + 1, canBeOrdered: true))
+                    }
+                    resultLines.append(Line(contents: "Missing key pairs:", indentationLevel: level, canBeOrdered: false, children: missingKeyPairs))
```

Comment:
`Missing key pairs:` ->
`Missing key: value pairs;`

so we end up having:

```
petAges:
|  Key Jethro:
|  |  Expected: 6
|  |  Actual  : 2
|  Missing key: value pairs;
|  |  Haddie: Optional(4)
|  Extra key: value pairs;
|  |  Henny: Optional(1)
```
instead of:

```
petAges:
|  Key Jethro:
|  |  Expected: 6
|  |  Actual  : 2
|  Missing key pairs:
|  |  Haddie: Optional(4)
|  Extra key pairs:
|  |  Henny: Optional(1)
```